atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/TransactionLocksManagerTest.java

```diff
@@ -156,14 +175,23 @@ public void throwsIfOnlyImmutableTimestampLockExpiredWhenCheckingBoth() {
     @Test
     public void doesNotCallLockRefresherIfNothingToCheck() {
         LockValidityChecker validityChecker = mock(LockValidityChecker.class);
-        ImmutableTimestampLockManager immutableTimestampLockManager =
-                new ImmutableTimestampLockManager(Optional.empty(), validityChecker);
+        TransactionLocksManager transactionLocksManager =
+                new TransactionLocksManager(Optional.empty(), validityChecker, unlocker);
 
-        assertThat(immutableTimestampLockManager.getExpiredImmutableTimestampAndCommitLocks(Optional.empty()))
+        assertThat(transactionLocksManager.getExpiredImmutableTimestampAndCommitLocks())
                 .isEmpty();
         verify(validityChecker, never()).getStillValidLockTokens(anySet());
     }
 
+    @Test
+    public void doesNotCallUnlockIfNoLocksToUnlock() {
+        TransactionLocksManager transactionLocksManager =
+                new TransactionLocksManager(Optional.empty(), validityChecker, unlocker);
+
+        transactionLocksManager.close();
+        verify(unlocker, never()).tryUnlock(anySet());
+    }
```

Comment:
there's probably a thing here where we should check that tryUnlock is called if the manager gets closed

Reply:
Good call, done